code/game/mecha/equipment/tools/tools.dm

```diff
@@ -530,102 +534,36 @@
 		target = get_turf(target)
 	if(!action_checks(target) || disabled || get_dist(chassis, target)>3)
 		return
-	//meh
+	var/obj/item/device/rcd/R = RCD
```

Comment:
Why do you need to cast this?

Reply:
I need `R` because the same block of code is used for controlling both the RPD and RCD. Which one gets controlled is determined by which one `R` is set equal to, which is determined by the `device` var.
It needs to be cast as `/obj/item/device/rcd` so I can access specific vars and procs on `R` such as `selected`.